internal/linker/link_node_modules.ts

```diff
@@ -111,16 +111,15 @@ async function resolveRoot(root: string|undefined, runfiles: Runfiles) {
   const fromManifest = runfiles.lookupDirectory(root);
   if (fromManifest) return fromManifest;
 
-  // Account for Bazel --legacy_external_runfiles
-  // which look like 'my_wksp/external/npm/node_modules'
-  if (await exists(path.join('external', root))) {
-    log_verbose('found legacy_external_runfiles, switching root to', path.join('external', root));
+  if (runfiles.execroot) {
```

Comment:
No. I think the two are orthogonal. Even with `legacy_external_runfiles` you get both external workspaces accessible via `../external_wksp` and via `./external/external_wksp`. This code just makes a distinction between being in the execroot for build rules where `./external/external_wksp` is the only resolution available and being in runfiles where `../external_wksp` is the preferred resolution since `./external/external_wksp` is only available with `legacy_external_runfiles`